src/AutoRest.CSharp/Common/Output/Models/OverloadMethodSignature.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using AutoRest.CSharp.Output.Models.Shared;
+
+namespace AutoRest.CSharp.Output.Models
+{
+    internal class OverloadMethodSignature
+    {
+        public MethodSignature MethodSignature { get; }
+        public MethodSignature PreviousMethodSignature { get; }
+        public IReadOnlyList<Parameter> MissingParameters { get; }
+        public FormattableString? Description { get; }
+        public bool IsHiddenFromUser { get; }
+
+        public OverloadMethodSignature(MethodSignature methodSignature, MethodSignature previousMethodSignature, IReadOnlyList<Parameter> missingParameters, FormattableString? description, bool isHiddenFromUser)
+        {
+            MethodSignature = methodSignature;
+            PreviousMethodSignature = previousMethodSignature;
+            MissingParameters = missingParameters;
+            Description = description;
+            IsHiddenFromUser = isHiddenFromUser;
+        }
+    }
```

Comment:
We could simply define a record for this

Reply:
and it would be better if we could have a `[DebuggerDisplay(...)]` attribute to customize the debugging output to show some useful information

Reply:
This can be a record, but I am not sure what should be set for `DebuggerDisplay`, it contains complex information of current generated method and previous GA contract method.

Reply:
Yeah it is a better to have.
If it is a class, when debugging, the debugger only shows the type name of this instance, which literally contains no information at all.
If we change this to a record, it will have a default `ToString` implementation to contain all its properties, which is better when debugging but it is usually too much.
Therefore I suggest we add a DebuggerDisplay to only show those information we find useful to show.
For instance. we could have:
OverloadMethodSignature({MethodName}, Missing: {MissingParameters})

Reply:
Don't think it's straightforward to put information to `DebuggerDisplay`, we would still need to look into the object since `MissingParameters` is a list of objects.